pkg/controllers/managementuser/secret/secret.go

```diff
@@ -157,8 +158,8 @@ func (c *ResourceSyncController) removeClusterIdFromSecretData(sec *corev1.Secre
 	return sec
 }
 
-func (c *ResourceSyncController) sync(key string, obj *corev1.Secret) (runtime.Object, error) {
-	if obj == nil {
+func (c *ResourceSyncController) sync(_ string, obj *corev1.Secret) (*corev1.Secret, error) {
```

Comment:
This controller is also problematic because it's modifying `obj` and returning it to cache without actually calling update on it. This is not something you introduced, but it is something we should fix.

Reply:
Good catch. I found it was only adding the annotation "provisioning.cattle.io/synced-at", and I couldn't find any usage of it (and a quick `kubectl get secret -A -o json | jq '.items.[]|.annotations."provisioning.cattle.io/synced-at"'` showed no occurrences), so I propose to just remove that unused annotation, see d3b69e44a4d862bcec51f8dd6c315c1fb9e18e70